test/WebJobs.Extensions.Tests/Extensions/Warmup/WarmupTriggerEndToEndTests.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Extensions.Tests.Common;
+using Microsoft.Azure.WebJobs.Extensions.Warmup.Trigger;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Host.Indexers;
+using Microsoft.Extensions.Hosting;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Warmup.Tests
+{
+    public class WarmupTriggerEndToEndTests
+    {
+        private static string _functionOut = null;
+        private readonly TestLoggerProvider _loggerProvider = new TestLoggerProvider();
+
+        [Theory]
+        [InlineData("WarmupTriggerParams.TestWarmup_String")]
+        [InlineData("WarmupTriggerParams.TestWarmup_JObject")]
+        [InlineData("WarmupTriggerParams.TestWarmup_Poco")]
+        public async Task WarmupTriggerTest_Success(string functionName)
+        {
+            _functionOut = null;
+            var hostStatus = new
```

Comment:
I think we need to define the actual input contract here (e.g. WarmupContext) not based on a loose bag of properties.